src/ngraph/runtime/gpu/cuda_emitter.cpp

```diff
@@ -231,6 +231,106 @@ size_t runtime::gpu::CUDAEmitter::build_pad(const runtime::gpu::GPURuntimeContex
     return primitive_index;
 }
 
+size_t runtime::gpu::CUDAEmitter::build_pad_dilation(const runtime::gpu::GPURuntimeContext* ctx,
```

Comment:
`build_pad` actually supports interior padding and so this kernel and the other perform the same function. 

One interesting point to consider, `build_pad` statically compiles the padding op and so doesn't require passing of input shapes/strides/padding etc to the kernel, whereas `build_pad_dilation` does the same thing but dynamically, passing in that information. It would be interesting to do a performance comparison between the two in the future. I have been wondering for some time if we can't get some performance boost from emitting shape information directly into the kernels (statically) at the cost of compiling a new kernel for every shape. 

My thought is that if there are many different shapes in a model, the compile time would become expensive and a dynamic kernel, such as `build_pad_dilation` would be better. On the other hand, if an op is called a lot with the same shape (e.g. LSTM?), the static version is likely faster as there are no memory accesses for kernel arguments.

Perhaps it is worth keeping `build_pad_dialation` and renaming it to `build_pad_dynamic` and `build_pad` to `build_pad_static`, so that we can test this out in the optimization pass. If not, at least keep it around in a branch.

Reply:
good point. Let's keep it here. We might need both for different purpose.